### DISCUSSION TITLE
Processing meteorological data to stream segments

### DISCUSSION THREAD

**Discussion by jds485** [USGS-R/drb-inland-salinity-ml#17] [General]
Based on conversations with Hayley Corson-Dosch @hcorson-dosch and Jake Zwart @jzwart

Met data for the PGDL streamflow and temperature models were processed to stream segments using 2 steps:
1. getting data from gridMET or other data source and matching to PRMS HRUs using Rich McDonald's code ([edited](https://github.com/hcorson-dosch/gridmetetl/blob/master/gridmetetl/etl.py) by Hayley), and then 
2. running PRMS-SNTemp using the HRU met data to generate segment-specific met data as outputs

This makes sense if the PRMS outputs are used for pretraining or as inputs to an ML model, but otherwise requires an extra model run to get the met data for stream segments. PRMS also adjusts the met data according to elevation and other factors, which may not be desirable for the ML model inputs. Jake suggests moving away from relying on step 2 and having a more general workflow that can go from gridded dataset to stream segments. 

Some next steps:
1. Check if HyTEST has tools available to match gridded data to any polygon, polyline, or point
2. Check in with Hayley for scripts that process the step 1 HRU gridMET data to PRMS segments (currently in jupyter notebooks on Hayley's computer). These scripts take gridMET parameters for each HRU and get area-weighted gridMET timeseries for PRMS segments, since many segments are associated with >1 HRU.
3. Check if using the [Geo Data Portal](https://cida.usgs.gov/gdp/) packages to gather and process gridMET data to HRUs is simpler/cleaner than the step 1 method above.

**Comment by jzwart** [comment]
@wdwatkins  also put  together some code for going from NOAA GEFS gridded drivers (netCDF) to stream segment points - see [targets here ](https://code.usgs.gov/wma/wp/drb-forecast-gefs-processing/-/blob/master/1_grid_to_segments.R)and [functions here ](https://code.usgs.gov/wma/wp/drb-forecast-gefs-processing/-/blob/master/1_grid_to_points/src/functions.R). Might be able to reuse some of that code if HyTEST doesn't have something or don't want to reuse Rich's code w/ Hayley's edits
